### ISSUE TITLE
Build failed in libcryptobox/blake2: Unknown gnu as macro parameter convention

### ISSUE THREAD

**Issue by djc** [vstakhov/rspamd#412]
I'm trying to package rspamd for Gentoo. Building it from the packaging system seems to work quite well, until it runs into this:

```
[ 84%] Building ASM object src/CMakeFiles/rspamd-server.dir/libcryptobox/blake2/x86-32.S.o
cd /var/tmp/portage/mail-filter/rspamd-1.0.7/work/rspamd-1.0.7_build/src && /usr/bin/i686-pc-linux-gnu-gcc  -DHAVE_CONFIG_H -O2 -march=native -pipe -ggdb3 -I/var/tmp/portage/mail-filter/rspamd-1.0.7/work/rspamd-1.0.7/src -I/var/tmp/portage/mail-filter/rspamd-1.0.7/work/rspamd-1.0.7/src/libutil -I/var/tmp/portage/mail-filter/rspamd-1.0.7/work/rspamd-1.0.7/src/libserver -I/var/tmp/portage/mail-filter/rspamd-1.0.7/work/rspamd-1.0.7/src/libmime -I/var/tmp/portage/mail-filter/rspamd-1.0.7/work/rspamd-1.0.7/src/libstat -I/var/tmp/portage/mail-filter/rspamd-1.0.7/work/rspamd-1.0.7/src/libcryptobox -I/var/tmp/portage/mail-filter/rspamd-1.0.7/work/rspamd-1.0.7/contrib/libucl -I/var/tmp/portage/mail-filter/rspamd-1.0.7/work/rspamd-1.0.7/contrib/uthash -I/var/tmp/portage/mail-filter/rspamd-1.0.7/work/rspamd-1.0.7/contrib/http-parser -I/var/tmp/portage/mail-filter/rspamd-1.0.7/work/rspamd-1.0.7/contrib/libottery -I/var/tmp/portage/mail-filter/rspamd-1.0.7/work/rspamd-1.0.7/contrib/xxhash -I/var/tmp/portage/mail-filter/rspamd-1.0.7/work/rspamd-1.0.7/contrib/cdb -I/var/tmp/portage/mail-filter/rspamd-1.0.7/work/rspamd-1.0.7/contrib/snowball/include -I/var/tmp/portage/mail-filter/rspamd-1.0.7/work/rspamd-1.0.7/contrib/librdns -I/var/tmp/portage/mail-filter/rspamd-1.0.7/work/rspamd-1.0.7/contrib/aho-corasick -I/var/tmp/portage/mail-filter/rspamd-1.0.7/work/rspamd-1.0.7_build/src -I/var/tmp/portage/mail-filter/rspamd-1.0.7/work/rspamd-1.0.7_build/src/libcryptobox -I/usr/include/luajit-2.0 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include -I/usr/include/gmime-2.6 -I/usr/include/hiredis -I/var/tmp/portage/mail-filter/rspamd-1.0.7/work/rspamd-1.0.7/compat    -o CMakeFiles/rspamd-server.dir/libcryptobox/blake2/x86-32.S.o -c /var/tmp/portage/mail-filter/rspamd-1.0.7/work/rspamd-1.0.7/src/libcryptobox/blake2/x86-32.S
In file included from /var/tmp/portage/mail-filter/rspamd-1.0.7/work/rspamd-1.0.7/src/libcryptobox/blake2/x86-32.S:1:0:
/var/tmp/portage/mail-filter/rspamd-1.0.7/work/rspamd-1.0.7/src/libcryptobox/blake2/../macro.S:6:3:error: #error Unknown gnu as macro parameter convention! Run ./configure
  #error Unknown gnu as macro parameter convention! Run ./configure
   ^
{standard input}: Assembler messages:
{standard input}: Error: .size expression for blake2b_blocks_x86 does not evaluate to a constant
{standard input}: Error: .size expression for _blake2b_blocks_x86 does not evaluate to a constant
src/CMakeFiles/rspamd-server.dir/build.make:2124: recipe for target 'src/CMakeFiles/rspamd-server.dir/libcryptobox/blake2/x86-32.S.o' failed
```

Built with rspamd-1.0.7. I'm happy to provide more info as needed.

**Comment by vstakhov** [comment]
Should be fixed in 1.0.8

**Comment by djc** [comment]
Yup, this works now. Thanks for the quick turnaround!
